### ISSUE TITLE
Lists of Figures and Tables do not appear in Table of Contents

### ISSUE THREAD

**Issue by mjost5v** [skulumani/thesis-gwu#37]
According to my correspondence with the ETD Administrator, the List of Figures and List of Tables should appear as entries in the Table of Contents. Currently, they do not. I've spent many hours trying to get them to render. The only workaround is to manually add them after the fact with the Professional version of Adobe Acrobat.

**Comment by skulumani** [comment]
No that's not correct way to do it.

The Lists of Figures and Tables were already part of the table of contents.
What changes were made to break that functionality?

**Comment by mjost5v** [comment]
I made no changes besides the spacing for which I put in a PR and included
some packages for my submission. Those packages would not cause this issue.
Your example also does not include the lists of figures and tables:
https://github.com/skulumani/thesis-gwu/blob/master/thesis_gwu_master.pdf

**Comment by skulumani** [comment]
Thanks for the details. There's been some changes to the class since 2018. Let me review and look into it.

**Comment by skulumani** [comment]
This commit

https://github.com/skulumani/thesis-gwu/commit/1f65226a8aff15a96b968d1fad37e4cf05a4bfc0

Has functional figures and tables in the TOC. 

The subsequent changes have broken the TOC.

**Comment by skulumani** [comment]
@mjost5v Thanks for bringing this up. I'm going to contact ETD directly and figure out the changes required between 2018 and 2020. 

Then I'll update this package to be current.

**Comment by themikesmith** [comment]
Hi all, I made the recent updates. Please find below verification from ETD
that the attached PDF conforms as of 2020.

> ---------- Forwarded message ---------
> From: etds etds <etds@gwu.edu>
> Date: Wed, Mar 4, 2020 at 1:23 PM
> Subject: Re: One more correction
> To: Michael Smith <mikesmith@gwmail.gwu.edu>
>
>
> Michael,
>
> This last one looks good to me.
>
> Valerie
>
@mjost5v <https://github.com/mjost5v> were you dealing with Valerie at ETD?
Did she make a mistake when she signed off on my template changes?
Best,
Mike

**Comment by mjost5v** [comment]
Yes, I was dealing with Valerie. She was very insistent about the lists
being in the table of contents.

**Comment by themikesmith** [comment]
That's incredibly frustrating. I specifically asked Valerie over a month
ago to verify my changes to the template before I contacted @skulumani -
Either she made a mistake on yours, or on my template, or she changed her
mind. (I hope it's not the third one.)

I just contacted Valerie as well. This is ridiculous. It's April 4th and
the deadline is the 15th.

For what it's worth @mjost5v my changes to the TOC involve the tocloft
package - I used this because (among other things) the TOC now needs to be
indented?

**Comment by skulumani** [comment]
My guess is that each of you are sending your own dissertation to check? She's obviously not consistent in her review of anyone's dissertation. 

The whole goal of this template is to allow for all of you to have a simple way to draft a thesis that conforms to the guidelines. It's now clear there are no guidelines. 

I also sent an email to Valerie at ETD to verify the required format of the sample thesis template. 

Hopefully she'll reply and I'll update the thesis, building off the work of @themikesmith, @mjost5v 

For now I've reverted to the 2018 version on master. 

I'm pretty sure the changes are just minor? Some line spacing, names, etc?

**Comment by mjost5v** [comment]
To be fair, the example from before had list of abbreviations listed 2x: 
https://github.com/skulumani/thesis-gwu/blob/643837c78e84b0e881a769861776e89674c9e2b7/thesis-sample.pdf

**Comment by skulumani** [comment]
@mjost5v  All of those things are configurable. 

You can utilize the `glossaries` or `acronym` package

i.e. `\hidelistofabbreviations` or `\hideglossarieslistofabbreviations`

**Comment by mjost5v** [comment]
My PR is a trivial line spacing change from 3 lines to 1 between the 
author and title for that section.

I did my best to get the lists into the TOC. After many hours of various 
LaTeX hacks, I was unable to do so and had to manually edit the 
generated PDF in Adobe Acrobat Professional and add the lines and 
bookmarks. I was fortunate to have Adobe CC already on my machine. SEAS 
students just got access to it this semester.

**Comment by themikesmith** [comment]
@mjost5v <https://github.com/mjost5v> , to fix your code and get them to
show up, the tocloft package is redefining the listoffigures commands
later, so....

Insert this in the class file replacing the calls to \listoffigures and
\listoftables at the bottom, in the group starting around line 1430 (there
is likely a more elegant solution here but given we have a very short time
to submit ...)

% Insert the list of figures.
 \newcommand{\mylistoffigures}{
    \listoffigures
    \phantomsection\addcontentsline{toc}{chapter}{List of Figures}
 }
 \if@gwu@listoffigures\mylistoffigures\fi %
 % Insert the list of tables.
 \newcommand{\mylistoftables}{
    \listoftables
    \phantomsection\addcontentsline{toc}{chapter}{List of Tables}
 }
 \if@gwu@listoftables\mylistoftables\fi %

**Comment by skulumani** [comment]
@mjost5v Thanks for the PR. I'll add it as well. 

I'm not super clear on the changes @themikesmith made but the current master has the version from 2018 that is fully functional and was once approved. 

My suggestion is to not worry about the formatting right now (unless it's due immediately or something) and just work on content. 

I'll fix the template once I hear back from Valerie. I'm not sure why this is so complicated for GWU/SEAS

**Comment by skulumani** [comment]
@themikesmith What was the reason for the `tocloft` package again?

**Comment by themikesmith** [comment]
Unfortunately our dissertations are due April 15. 

@skulumani I added the tocloft package because apparently the TOC needs to be indented? That seemed like the most elegant way to do it while also allowing for random other things like controlling for line spacing, etc., but I'm not married to it. @mjost5v See above for a (very hacky) fix given the deadline.

In terms of other changes, I'm sorry I don't recall exactly what they were - I put this out of mind a month ago when Valerie approved the template updates. Not an exclusive list, but beyond the above, I definitely futzed with the formatting and line spacing and numerous other minutiae.

I can help you update this after April 15.

**Comment by themikesmith** [comment]
FYI @mjost5v : tocloft is also likely overriding the other `\listofX` functions, so the hack given the deadline for those will be similar

**Comment by skulumani** [comment]
No problem @themikesmith. I appreciate the help. Hopefully you and @mjost5v can get something helpful in order to submit on time.

I'll work to ensure it's updated. 

For anyone else, make sure you utilize this class file from source rather than through CTAN/Overleaf as though are now probably not valid anymore.

**Comment by themikesmith** [comment]
I just updated the template on overleaf with directions on how to fix if they encounter this temporary problem given the 4/15 deadline @skulumani

**Comment by mjost5v** [comment]
@skulumani and @themikesmith, I manually edited my output in Adobe 
Acrobat Professional to get the thesis updated and approved for 
publication yesterday.

**Comment by themikesmith** [comment]
@mjost5v Congratulations!! I'm a week or so behind you :) I'm sorry you had to deal with what seems like fickle guidelines. Best of luck.
